azext_edge/edge/providers/orchestration/resources/instances.py

```diff
@@ -346,24 +355,69 @@ def disable_secretsync(
-                spc_poller = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.begin_delete(
-                    resource_group_name=resource_group_name,
-                    azure_key_vault_secret_provider_class_name=secretsync_spc["name"],
-                )
-                wait_for_terminal_state(spc_poller, **kwargs)
+            secretsync_spcs = self._find_existing_resources(
+                cl_resources=cl_resources, resource_type=SPC_RESOURCE_TYPE
+            )
+            secretsyncs = self._find_existing_resources(
+                cl_resources=cl_resources, resource_type=SECRET_SYNC_RESOURCE_TYPE
+            )
+
+            related_secretsyncs = []
+            if secretsync_spcs:
+                for secretsync_spc in secretsync_spcs:
+                    spc_poller = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.begin_delete(
+                        resource_group_name=resource_group_name,
+                        azure_key_vault_secret_provider_class_name=secretsync_spc["name"],
+                    )
+                    wait_for_terminal_state(spc_poller, **kwargs)
+
+                    # get associated secret sync names
+                    related_secretsyncs.extend(
+                        self._find_spc_related_secretsyncs(
+                            spc_name=secretsync_spc["name"],
+                            secretsync_resources=secretsyncs,
+                        )
+                    )
+
+                # delete associated secret syncs
+                if related_secretsyncs:
+                    for secretsync in related_secretsyncs:
+                        secretsync_poller = self.ssc_mgmt_client.secret_syncs.begin_delete(
+                            resource_group_name=resource_group_name,
+                            secret_sync_name=secretsync,
+                        )
+                        wait_for_terminal_state(secretsync_poller, **kwargs)
+
                 return
         logger.warning(f"No secret provider class detected.\n{get_enable_syntax(name, resource_group_name)}")
 
-    def _find_existing_spc(self, cl_resources: List[dict]) -> Optional[dict]:
+    def _find_existing_resources(self, cl_resources: List[dict], resource_type: str) -> Optional[List[dict]]:
+        resources = []
         for resource in cl_resources:
-            if resource["type"].lower() == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
+            if resource["type"].lower() == resource_type:
                 resource_id_container = parse_resource_id(resource["id"])
-                return self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
-                    resource_group_name=resource_id_container.resource_group_name,
-                    azure_key_vault_secret_provider_class_name=resource_id_container.resource_name,
-                )
+                if resource_type == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
```

Comment:
Shouldn't we reference the module constants that were added ?

Reply:
will update!